pybigquery/geography.py

```diff
@@ -0,0 +1,160 @@
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import geoalchemy2
+from geoalchemy2.shape import to_shape
+import geoalchemy2.functions
+from shapely import wkb, wkt
+import sqlalchemy.ext.compiler
+from sqlalchemy.sql.elements import BindParameter
+
+SRID = 4326  # WGS84
+
+
+class WKB(geoalchemy2.WKBElement):
+
+    geom_from_extended_version = "ST_GeogFromWKB"
+
+    def __init__(self, data, srid=SRID, extended=True):
+        if srid != SRID:
+            raise AssertionError("Bad srid", srid)
+        if not extended:
+            raise AssertionError("Extended must be True.")
+        super().__init__(data, srid, True)
+
+    @property
+    def wkt(self):
+        return WKT(to_shape(self).wkt)
+
+
+class WKT(geoalchemy2.WKTElement):
+
+    geom_from_extended_version = "ST_GeogFromText"
+
+    def __init__(self, data):
+        super().__init__(data, SRID, True)
+
+    @property
+    def wkb(self):
+        return WKB(wkb.dumps(wkt.loads(self.data)))
+
+
+class GEOGRAPHY(geoalchemy2.Geography):
+
+    ElementType = WKB
+
+    def __init__(self):
+        super().__init__(
+            geometry_type=None, spatial_index=False, srid=SRID,
+        )
+        self.extended = True
+
+    # Un-inherit the bind function that adds an ST_GeogFromText.
+    # It's unnecessary and causes BigQuery to error.
+    #
+    # Some things to note about this:
+    #
+    # 1. bind_expression can't always know the value.  When multiple
+    #    rows are being inserted, the values may be different in each
+    #    row.  As a consequence, we have to treat all the values as WKT.
+    #
+    # 2. This applies equally to explicitly converting with
+    #    st_geogfromtext, or implicitly with the geography parameter
+    #    conversion.
+    #
+    # 3. We handle different types using bind_processor, below.
+    #
+    bind_expression = sqlalchemy.sql.type_api.TypeEngine.bind_expression
+
+    def bind_processor(self, dialect):
+        def process(bindvalue):
+            if isinstance(bindvalue, WKT):
+                return bindvalue.data
+            elif isinstance(bindvalue, WKB):
+                return bindvalue.wkt.data
+            else:
+                return bindvalue
+
+        return process
+
+
+@sqlalchemy.ext.compiler.compiles(geoalchemy2.functions.GenericFunction, "bigquery")
+def _fixup_st_arguments(element, compiler, **kw):
```

Comment:
Could we get a docstring here? I'd like some more context about how this function is expected to be used.